objectivec/cxx_utils.h

```diff
@@ -0,0 +1,32 @@
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+
+#include <optional>
+#include <string>
+#include <variant>
+
+NS_ASSUME_NONNULL_BEGIN
+@class ORTValue;
+
+@interface ORTUtils : NSObject
+
++ (NSString*)toNSString:(const std::string&)str;
++ (nullable NSString*)toNullableNSString:(const std::optional<std::basic_string<char>>&)str;
+
++ (std::string)toStdString:(NSString*)str;
++ (std::optional<std::basic_string<char>>)toStdOptionalString:(nullable NSString*)str;
+
++ (std::vector<std::string>)toStdStringVector:(NSArray<NSString*>*)strs;
++ (NSArray<NSString*>*)toNSStringNSArray:(const std::vector<std::string>&)strs;
+
++ (nullable NSArray<ORTValue*>*)toORTValueNSArray:(const std::vector<OrtValue*>&)values
+                                            error:(NSError**)error;
```

Comment:
as this conversion is a bit more complicated with the ownership transfer, consider a name like this: `wrapUnownedCAPIOrtValues`.

and for `toOrtValueVector`, something like `getWrappedCAPIOrtValues`.

at least the ownership semantics of these OrtValue functions should be documented if they are not obvious from the name.

there are a few "ort value" types used so it helps to be explicit when referring to them, like "CAPIOrtValue".